{"cells":[{"cell_type":"markdown","metadata":{"id":"sKoJB9-u26Je"},"source":["# Data Preprocessing"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"fxo47XtD2V-D","outputId":"533e8741-05b3-4d9a-dfc8-cf6696867518"},"outputs":[{"name":"stdout","output_type":"stream","text":["/kaggle/input/genomics-of-drug-sensitivity-in-cancer-gdsc/Cell_Lines_Details.xlsx\n","/kaggle/input/genomics-of-drug-sensitivity-in-cancer-gdsc/GDSC_DATASET.csv\n","/kaggle/input/genomics-of-drug-sensitivity-in-cancer-gdsc/GDSC2-dataset.csv\n","/kaggle/input/genomics-of-drug-sensitivity-in-cancer-gdsc/Compounds-annotation.csv\n"]}],"source":["# This Python 3 environment comes with many helpful analytics libraries installed\n","# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python\n","# For example, here's several helpful packages to load\n","\n","import numpy as np # linear algebra\n","import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)\n","\n","# Input data files are available in the read-only \"../input/\" directory\n","# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory\n","\n","import os\n","for dirname, _, filenames in os.walk('/kaggle/input'):\n","    for filename in filenames:\n","        print(os.path.join(dirname, filename))\n","\n","# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using \"Save & Run All\"\n","# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"92Gg2oBG2V-D","outputId":"96c2c854-ad31-482e-c150-31d3f5a51e34"},"outputs":[{"name":"stderr","output_type":"stream","text":["/opt/conda/lib/python3.10/site-packages/openpyxl/worksheet/_reader.py:329: UserWarning: Unknown extension is not supported and will be removed\n","  warn(msg)\n"]}],"source":["# Load the datasets\n","gdsc_dataset = pd.read_csv('/kaggle/input/genomics-of-drug-sensitivity-in-cancer-gdsc/GDSC_DATASET.csv')\n","compounds_annotation = pd.read_csv('/kaggle/input/genomics-of-drug-sensitivity-in-cancer-gdsc/Compounds-annotation.csv')\n","gdsc2_dataset = pd.read_csv('/kaggle/input/genomics-of-drug-sensitivity-in-cancer-gdsc/GDSC2-dataset.csv')\n","cell_lines_details = pd.read_excel('/kaggle/input/genomics-of-drug-sensitivity-in-cancer-gdsc/Cell_Lines_Details.xlsx')"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"IdlCp_Mo2V-E","outputId":"36c6def1-0cb4-4b3a-c089-50784fe5dde3"},"outputs":[{"data":{"text/html":"<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>COSMIC_ID</th>\n      <th>CELL_LINE_NAME</th>\n      <th>TCGA_DESC</th>\n      <th>DRUG_ID</th>\n      <th>DRUG_NAME</th>\n      <th>LN_IC50</th>\n      <th>AUC</th>\n      <th>Z_SCORE</th>\n      <th>GDSC Tissue descriptor 1</th>\n      <th>GDSC Tissue descriptor 2</th>\n      <th>Cancer Type (matching TCGA label)</th>\n      <th>Microsatellite instability Status (MSI)</th>\n      <th>Screen Medium</th>\n      <th>Growth Properties</th>\n      <th>CNA</th>\n      <th>Gene Expression</th>\n      <th>Methylation</th>\n      <th>TARGET</th>\n      <th>TARGET_PATHWAY</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>683667</td>\n      <td>PFSK-1</td>\n      <td>MB</td>\n      <td>1003</td>\n      <td>Camptothecin</td>\n      <td>-1.463887</td>\n      <td>0.930220</td>\n      <td>0.433123</td>\n      <td>nervous_system</td>\n      <td>medulloblastoma</td>\n      <td>MB</td>\n      <td>MSS/MSI-L</td>\n      <td>R</td>\n      <td>Adherent</td>\n      <td>Y</td>\n      <td>Y</td>\n      <td>Y</td>\n      <td>TOP1</td>\n      <td>DNA replication</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>684057</td>\n      <td>ES5</td>\n      <td>UNCLASSIFIED</td>\n      <td>1003</td>\n      <td>Camptothecin</td>\n      <td>-3.360586</td>\n      <td>0.791072</td>\n      <td>-0.599569</td>\n      <td>bone</td>\n      <td>ewings_sarcoma</td>\n      <td>NaN</td>\n      <td>MSS/MSI-L</td>\n      <td>R</td>\n      <td>Adherent</td>\n      <td>Y</td>\n      <td>Y</td>\n      <td>Y</td>\n      <td>TOP1</td>\n      <td>DNA replication</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>684059</td>\n      <td>ES7</td>\n      <td>UNCLASSIFIED</td>\n      <td>1003</td>\n      <td>Camptothecin</td>\n      <td>-5.044940</td>\n      <td>0.592660</td>\n      <td>-1.516647</td>\n      <td>bone</td>\n      <td>ewings_sarcoma</td>\n      <td>NaN</td>\n      <td>MSS/MSI-L</td>\n      <td>R</td>\n      <td>Adherent</td>\n      <td>Y</td>\n      <td>Y</td>\n      <td>Y</td>\n      <td>TOP1</td>\n      <td>DNA replication</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>684062</td>\n      <td>EW-11</td>\n      <td>UNCLASSIFIED</td>\n      <td>1003</td>\n      <td>Camptothecin</td>\n      <td>-3.741991</td>\n      <td>0.734047</td>\n      <td>-0.807232</td>\n      <td>bone</td>\n      <td>ewings_sarcoma</td>\n      <td>NaN</td>\n      <td>MSS/MSI-L</td>\n      <td>R</td>\n      <td>Adherent</td>\n      <td>Y</td>\n      <td>Y</td>\n      <td>Y</td>\n      <td>TOP1</td>\n      <td>DNA replication</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>684072</td>\n      <td>SK-ES-1</td>\n      <td>UNCLASSIFIED</td>\n      <td>1003</td>\n      <td>Camptothecin</td>\n      <td>-5.142961</td>\n      <td>0.582439</td>\n      <td>-1.570016</td>\n      <td>bone</td>\n      <td>ewings_sarcoma</td>\n      <td>NaN</td>\n      <td>MSS/MSI-L</td>\n      <td>R</td>\n      <td>Semi-Adherent</td>\n      <td>Y</td>\n      <td>Y</td>\n      <td>Y</td>\n      <td>TOP1</td>\n      <td>DNA replication</td>\n    </tr>\n  </tbody>\n</table>\n</div>","text/plain":"   COSMIC_ID CELL_LINE_NAME     TCGA_DESC  DRUG_ID     DRUG_NAME   LN_IC50  \\\n0     683667         PFSK-1            MB     1003  Camptothecin -1.463887   \n1     684057            ES5  UNCLASSIFIED     1003  Camptothecin -3.360586   \n2     684059            ES7  UNCLASSIFIED     1003  Camptothecin -5.044940   \n3     684062          EW-11  UNCLASSIFIED     1003  Camptothecin -3.741991   \n4     684072        SK-ES-1  UNCLASSIFIED     1003  Camptothecin -5.142961   \n\n        AUC   Z_SCORE GDSC Tissue descriptor 1 GDSC Tissue descriptor 2  \\\n0  0.930220  0.433123           nervous_system          medulloblastoma   \n1  0.791072 -0.599569                     bone           ewings_sarcoma   \n2  0.592660 -1.516647                     bone           ewings_sarcoma   \n3  0.734047 -0.807232                     bone           ewings_sarcoma   \n4  0.582439 -1.570016                     bone           ewings_sarcoma   \n\n  Cancer Type (matching TCGA label) Microsatellite instability Status (MSI)  \\\n0                                MB                               MSS/MSI-L   \n1                               NaN                               MSS/MSI-L   \n2                               NaN                               MSS/MSI-L   \n3                               NaN                               MSS/MSI-L   \n4                               NaN                               MSS/MSI-L   \n\n  Screen Medium Growth Properties CNA Gene Expression Methylation TARGET  \\\n0             R          Adherent   Y               Y           Y   TOP1   \n1             R          Adherent   Y               Y           Y   TOP1   \n2             R          Adherent   Y               Y           Y   TOP1   \n3             R          Adherent   Y               Y           Y   TOP1   \n4             R     Semi-Adherent   Y               Y           Y   TOP1   \n\n    TARGET_PATHWAY  \n0  DNA replication  \n1  DNA replication  \n2  DNA replication  \n3  DNA replication  \n4  DNA replication  "},"metadata":{},"output_type":"display_data"},{"data":{"text/html":"<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>DRUG_ID</th>\n      <th>SCREENING_SITE</th>\n      <th>DRUG_NAME</th>\n      <th>SYNONYMS</th>\n      <th>TARGET</th>\n      <th>TARGET_PATHWAY</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>1</td>\n      <td>MGH</td>\n      <td>Erlotinib</td>\n      <td>Tarceva, RG-1415, CP-358774, OSI-774, Ro-50823...</td>\n      <td>EGFR</td>\n      <td>EGFR signaling</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>3</td>\n      <td>MGH</td>\n      <td>Rapamycin</td>\n      <td>AY-22989, Sirolimus, WY-090217, Torisel, Rapamune</td>\n      <td>MTORC1</td>\n      <td>PI3K/MTOR signaling</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>5</td>\n      <td>MGH</td>\n      <td>Sunitinib</td>\n      <td>Sutent, Sunitinib Malate, SU-11248</td>\n      <td>PDGFR, KIT, VEGFR, FLT3, RET, CSF1R</td>\n      <td>RTK signaling</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>6</td>\n      <td>MGH</td>\n      <td>PHA-665752</td>\n      <td>PHA665752, PHA 665752</td>\n      <td>MET</td>\n      <td>RTK signaling</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>9</td>\n      <td>MGH</td>\n      <td>MG-132</td>\n      <td>LLL cpd, MG 132, MG132</td>\n      <td>Proteasome, CAPN1</td>\n      <td>Protein stability and degradation</td>\n    </tr>\n  </tbody>\n</table>\n</div>","text/plain":"   DRUG_ID SCREENING_SITE   DRUG_NAME  \\\n0        1            MGH   Erlotinib   \n1        3            MGH   Rapamycin   \n2        5            MGH   Sunitinib   \n3        6            MGH  PHA-665752   \n4        9            MGH      MG-132   \n\n                                            SYNONYMS  \\\n0  Tarceva, RG-1415, CP-358774, OSI-774, Ro-50823...   \n1  AY-22989, Sirolimus, WY-090217, Torisel, Rapamune   \n2                 Sutent, Sunitinib Malate, SU-11248   \n3                              PHA665752, PHA 665752   \n4                             LLL cpd, MG 132, MG132   \n\n                                TARGET                     TARGET_PATHWAY  \n0                                 EGFR                     EGFR signaling  \n1                               MTORC1                PI3K/MTOR signaling  \n2  PDGFR, KIT, VEGFR, FLT3, RET, CSF1R                      RTK signaling  \n3                                  MET                      RTK signaling  \n4                    Proteasome, CAPN1  Protein stability and degradation  "},"metadata":{},"output_type":"display_data"},{"data":{"text/html":"<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>DATASET</th>\n      <th>NLME_RESULT_ID</th>\n      <th>NLME_CURVE_ID</th>\n      <th>COSMIC_ID</th>\n      <th>CELL_LINE_NAME</th>\n      <th>SANGER_MODEL_ID</th>\n      <th>TCGA_DESC</th>\n      <th>DRUG_ID</th>\n      <th>DRUG_NAME</th>\n      <th>PUTATIVE_TARGET</th>\n      <th>PATHWAY_NAME</th>\n      <th>COMPANY_ID</th>\n      <th>WEBRELEASE</th>\n      <th>MIN_CONC</th>\n      <th>MAX_CONC</th>\n      <th>LN_IC50</th>\n      <th>AUC</th>\n      <th>RMSE</th>\n      <th>Z_SCORE</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>GDSC2</td>\n      <td>343</td>\n      <td>15946310</td>\n      <td>683667</td>\n      <td>PFSK-1</td>\n      <td>SIDM01132</td>\n      <td>MB</td>\n      <td>1003</td>\n      <td>Camptothecin</td>\n      <td>TOP1</td>\n      <td>DNA replication</td>\n      <td>1046</td>\n      <td>Y</td>\n      <td>0.0001</td>\n      <td>0.1</td>\n      <td>-1.463887</td>\n      <td>0.930220</td>\n      <td>0.089052</td>\n      <td>0.433123</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>GDSC2</td>\n      <td>343</td>\n      <td>15946548</td>\n      <td>684052</td>\n      <td>A673</td>\n      <td>SIDM00848</td>\n      <td>UNCLASSIFIED</td>\n      <td>1003</td>\n      <td>Camptothecin</td>\n      <td>TOP1</td>\n      <td>DNA replication</td>\n      <td>1046</td>\n      <td>Y</td>\n      <td>0.0001</td>\n      <td>0.1</td>\n      <td>-4.869455</td>\n      <td>0.614970</td>\n      <td>0.111351</td>\n      <td>-1.421100</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>GDSC2</td>\n      <td>343</td>\n      <td>15946830</td>\n      <td>684057</td>\n      <td>ES5</td>\n      <td>SIDM00263</td>\n      <td>UNCLASSIFIED</td>\n      <td>1003</td>\n      <td>Camptothecin</td>\n      <td>TOP1</td>\n      <td>DNA replication</td>\n      <td>1046</td>\n      <td>Y</td>\n      <td>0.0001</td>\n      <td>0.1</td>\n      <td>-3.360586</td>\n      <td>0.791072</td>\n      <td>0.142855</td>\n      <td>-0.599569</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>GDSC2</td>\n      <td>343</td>\n      <td>15947087</td>\n      <td>684059</td>\n      <td>ES7</td>\n      <td>SIDM00269</td>\n      <td>UNCLASSIFIED</td>\n      <td>1003</td>\n      <td>Camptothecin</td>\n      <td>TOP1</td>\n      <td>DNA replication</td>\n      <td>1046</td>\n      <td>Y</td>\n      <td>0.0001</td>\n      <td>0.1</td>\n      <td>-5.044940</td>\n      <td>0.592660</td>\n      <td>0.135539</td>\n      <td>-1.516647</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>GDSC2</td>\n      <td>343</td>\n      <td>15947369</td>\n      <td>684062</td>\n      <td>EW-11</td>\n      <td>SIDM00203</td>\n      <td>UNCLASSIFIED</td>\n      <td>1003</td>\n      <td>Camptothecin</td>\n      <td>TOP1</td>\n      <td>DNA replication</td>\n      <td>1046</td>\n      <td>Y</td>\n      <td>0.0001</td>\n      <td>0.1</td>\n      <td>-3.741991</td>\n      <td>0.734047</td>\n      <td>0.128059</td>\n      <td>-0.807232</td>\n    </tr>\n  </tbody>\n</table>\n</div>","text/plain":"  DATASET  NLME_RESULT_ID  NLME_CURVE_ID  COSMIC_ID CELL_LINE_NAME  \\\n0   GDSC2             343       15946310     683667         PFSK-1   \n1   GDSC2             343       15946548     684052           A673   \n2   GDSC2             343       15946830     684057            ES5   \n3   GDSC2             343       15947087     684059            ES7   \n4   GDSC2             343       15947369     684062          EW-11   \n\n  SANGER_MODEL_ID     TCGA_DESC  DRUG_ID     DRUG_NAME PUTATIVE_TARGET  \\\n0       SIDM01132            MB     1003  Camptothecin            TOP1   \n1       SIDM00848  UNCLASSIFIED     1003  Camptothecin            TOP1   \n2       SIDM00263  UNCLASSIFIED     1003  Camptothecin            TOP1   \n3       SIDM00269  UNCLASSIFIED     1003  Camptothecin            TOP1   \n4       SIDM00203  UNCLASSIFIED     1003  Camptothecin            TOP1   \n\n      PATHWAY_NAME  COMPANY_ID WEBRELEASE  MIN_CONC  MAX_CONC   LN_IC50  \\\n0  DNA replication        1046          Y    0.0001       0.1 -1.463887   \n1  DNA replication        1046          Y    0.0001       0.1 -4.869455   \n2  DNA replication        1046          Y    0.0001       0.1 -3.360586   \n3  DNA replication        1046          Y    0.0001       0.1 -5.044940   \n4  DNA replication        1046          Y    0.0001       0.1 -3.741991   \n\n        AUC      RMSE   Z_SCORE  \n0  0.930220  0.089052  0.433123  \n1  0.614970  0.111351 -1.421100  \n2  0.791072  0.142855 -0.599569  \n3  0.592660  0.135539 -1.516647  \n4  0.734047  0.128059 -0.807232  "},"metadata":{},"output_type":"display_data"},{"data":{"text/html":"<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>Sample Name</th>\n      <th>COSMIC identifier</th>\n      <th>Whole Exome Sequencing (WES)</th>\n      <th>Copy Number Alterations (CNA)</th>\n      <th>Gene Expression</th>\n      <th>Methylation</th>\n      <th>Drug\\nResponse</th>\n      <th>GDSC\\nTissue descriptor 1</th>\n      <th>GDSC\\nTissue\\ndescriptor 2</th>\n      <th>Cancer Type\\n(matching TCGA label)</th>\n      <th>Microsatellite \\ninstability Status (MSI)</th>\n      <th>Screen Medium</th>\n      <th>Growth Properties</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>A253</td>\n      <td>906794.0</td>\n      <td>Y</td>\n      <td>Y</td>\n      <td>Y</td>\n      <td>Y</td>\n      <td>Y</td>\n      <td>aero_dig_tract</td>\n      <td>head and neck</td>\n      <td>NaN</td>\n      <td>MSS/MSI-L</td>\n      <td>D/F12</td>\n      <td>Adherent</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>BB30-HNC</td>\n      <td>753531.0</td>\n      <td>Y</td>\n      <td>Y</td>\n      <td>Y</td>\n      <td>Y</td>\n      <td>Y</td>\n      <td>aero_dig_tract</td>\n      <td>head and neck</td>\n      <td>HNSC</td>\n      <td>MSS/MSI-L</td>\n      <td>D/F12</td>\n      <td>Adherent</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>BB49-HNC</td>\n      <td>753532.0</td>\n      <td>Y</td>\n      <td>Y</td>\n      <td>Y</td>\n      <td>Y</td>\n      <td>Y</td>\n      <td>aero_dig_tract</td>\n      <td>head and neck</td>\n      <td>HNSC</td>\n      <td>MSS/MSI-L</td>\n      <td>D/F12</td>\n      <td>Adherent</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>BHY</td>\n      <td>753535.0</td>\n      <td>Y</td>\n      <td>Y</td>\n      <td>Y</td>\n      <td>Y</td>\n      <td>Y</td>\n      <td>aero_dig_tract</td>\n      <td>head and neck</td>\n      <td>HNSC</td>\n      <td>MSS/MSI-L</td>\n      <td>D/F12</td>\n      <td>Adherent</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>BICR10</td>\n      <td>1290724.0</td>\n      <td>Y</td>\n      <td>Y</td>\n      <td>Y</td>\n      <td>Y</td>\n      <td>Y</td>\n      <td>aero_dig_tract</td>\n      <td>head and neck</td>\n      <td>HNSC</td>\n      <td>MSS/MSI-L</td>\n      <td>D/F12</td>\n      <td>Adherent</td>\n    </tr>\n  </tbody>\n</table>\n</div>","text/plain":"  Sample Name  COSMIC identifier Whole Exome Sequencing (WES)  \\\n0        A253           906794.0                            Y   \n1    BB30-HNC           753531.0                            Y   \n2    BB49-HNC           753532.0                            Y   \n3         BHY           753535.0                            Y   \n4      BICR10          1290724.0                            Y   \n\n  Copy Number Alterations (CNA) Gene Expression Methylation Drug\\nResponse  \\\n0                             Y               Y           Y              Y   \n1                             Y               Y           Y              Y   \n2                             Y               Y           Y              Y   \n3                             Y               Y           Y              Y   \n4                             Y               Y           Y              Y   \n\n  GDSC\\nTissue descriptor 1 GDSC\\nTissue\\ndescriptor 2  \\\n0            aero_dig_tract              head and neck   \n1            aero_dig_tract              head and neck   \n2            aero_dig_tract              head and neck   \n3            aero_dig_tract              head and neck   \n4            aero_dig_tract              head and neck   \n\n  Cancer Type\\n(matching TCGA label)  \\\n0                                NaN   \n1                               HNSC   \n2                               HNSC   \n3                               HNSC   \n4                               HNSC   \n\n  Microsatellite \\ninstability Status (MSI) Screen Medium Growth Properties  \n0                                 MSS/MSI-L         D/F12          Adherent  \n1                                 MSS/MSI-L         D/F12          Adherent  \n2                                 MSS/MSI-L         D/F12          Adherent  \n3                                 MSS/MSI-L         D/F12          Adherent  \n4                                 MSS/MSI-L         D/F12          Adherent  "},"metadata":{},"output_type":"display_data"}],"source":["display(gdsc_dataset.head())\n","display(compounds_annotation.head())\n","display(gdsc2_dataset.head())\n","display(cell_lines_details.head())"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"sxlIOiAJ2V-E","outputId":"0876d195-3e0d-4033-d5e8-d6f03cb2d3c3"},"outputs":[{"data":{"text/html":"<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>COSMIC_ID</th>\n      <th>PUTATIVE_TARGET</th>\n      <th>PATHWAY_NAME</th>\n      <th>MIN_CONC</th>\n      <th>MAX_CONC</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>683667</td>\n      <td>TOP1</td>\n      <td>DNA replication</td>\n      <td>0.0001</td>\n      <td>0.1</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>684052</td>\n      <td>TOP1</td>\n      <td>DNA replication</td>\n      <td>0.0001</td>\n      <td>0.1</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>684057</td>\n      <td>TOP1</td>\n      <td>DNA replication</td>\n      <td>0.0001</td>\n      <td>0.1</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>684059</td>\n      <td>TOP1</td>\n      <td>DNA replication</td>\n      <td>0.0001</td>\n      <td>0.1</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>684062</td>\n      <td>TOP1</td>\n      <td>DNA replication</td>\n      <td>0.0001</td>\n      <td>0.1</td>\n    </tr>\n  </tbody>\n</table>\n</div>","text/plain":"   COSMIC_ID PUTATIVE_TARGET     PATHWAY_NAME  MIN_CONC  MAX_CONC\n0     683667            TOP1  DNA replication    0.0001       0.1\n1     684052            TOP1  DNA replication    0.0001       0.1\n2     684057            TOP1  DNA replication    0.0001       0.1\n3     684059            TOP1  DNA replication    0.0001       0.1\n4     684062            TOP1  DNA replication    0.0001       0.1"},"metadata":{},"output_type":"display_data"},{"data":{"text/html":"<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>COSMIC_ID</th>\n      <th>GDSC Tissue descriptor 1</th>\n      <th>GDSC Tissue descriptor 2</th>\n      <th>Cancer Type (matching TCGA label)</th>\n      <th>Microsatellite instability Status (MSI)</th>\n      <th>Screen Medium</th>\n      <th>CNA</th>\n      <th>Methylation</th>\n      <th>TARGET_PATHWAY</th>\n      <th>DRUG_NAME</th>\n      <th>LN_IC50</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>683667</td>\n      <td>nervous_system</td>\n      <td>medulloblastoma</td>\n      <td>MB</td>\n      <td>MSS/MSI-L</td>\n      <td>R</td>\n      <td>Y</td>\n      <td>Y</td>\n      <td>DNA replication</td>\n      <td>Camptothecin</td>\n      <td>-1.463887</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>684057</td>\n      <td>bone</td>\n      <td>ewings_sarcoma</td>\n      <td>NaN</td>\n      <td>MSS/MSI-L</td>\n      <td>R</td>\n      <td>Y</td>\n      <td>Y</td>\n      <td>DNA replication</td>\n      <td>Camptothecin</td>\n      <td>-3.360586</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>684059</td>\n      <td>bone</td>\n      <td>ewings_sarcoma</td>\n      <td>NaN</td>\n      <td>MSS/MSI-L</td>\n      <td>R</td>\n      <td>Y</td>\n      <td>Y</td>\n      <td>DNA replication</td>\n      <td>Camptothecin</td>\n      <td>-5.044940</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>684062</td>\n      <td>bone</td>\n      <td>ewings_sarcoma</td>\n      <td>NaN</td>\n      <td>MSS/MSI-L</td>\n      <td>R</td>\n      <td>Y</td>\n      <td>Y</td>\n      <td>DNA replication</td>\n      <td>Camptothecin</td>\n      <td>-3.741991</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>684072</td>\n      <td>bone</td>\n      <td>ewings_sarcoma</td>\n      <td>NaN</td>\n      <td>MSS/MSI-L</td>\n      <td>R</td>\n      <td>Y</td>\n      <td>Y</td>\n      <td>DNA replication</td>\n      <td>Camptothecin</td>\n      <td>-5.142961</td>\n    </tr>\n  </tbody>\n</table>\n</div>","text/plain":"   COSMIC_ID GDSC Tissue descriptor 1 GDSC Tissue descriptor 2  \\\n0     683667           nervous_system          medulloblastoma   \n1     684057                     bone           ewings_sarcoma   \n2     684059                     bone           ewings_sarcoma   \n3     684062                     bone           ewings_sarcoma   \n4     684072                     bone           ewings_sarcoma   \n\n  Cancer Type (matching TCGA label) Microsatellite instability Status (MSI)  \\\n0                                MB                               MSS/MSI-L   \n1                               NaN                               MSS/MSI-L   \n2                               NaN                               MSS/MSI-L   \n3                               NaN                               MSS/MSI-L   \n4                               NaN                               MSS/MSI-L   \n\n  Screen Medium CNA Methylation   TARGET_PATHWAY     DRUG_NAME   LN_IC50  \n0             R   Y           Y  DNA replication  Camptothecin -1.463887  \n1             R   Y           Y  DNA replication  Camptothecin -3.360586  \n2             R   Y           Y  DNA replication  Camptothecin -5.044940  \n3             R   Y           Y  DNA replication  Camptothecin -3.741991  \n4             R   Y           Y  DNA replication  Camptothecin -5.142961  "},"metadata":{},"output_type":"display_data"}],"source":["gdsc2_subset = gdsc2_dataset[['COSMIC_ID', 'PUTATIVE_TARGET', 'PATHWAY_NAME', 'MIN_CONC', 'MAX_CONC']]\n","display(gdsc2_subset.head())\n","gdsc_subset = gdsc_dataset[['COSMIC_ID',\n","                            'GDSC Tissue descriptor 1',\n","                            'GDSC Tissue descriptor 2',\n","                            'Cancer Type (matching TCGA label)',\n","                            'Microsatellite instability Status (MSI)',\n","                            'Screen Medium',\n","                            'CNA',\n","                            'Methylation',\n","                            'TARGET_PATHWAY',\n","                            'DRUG_NAME',\n","                            'LN_IC50'\n","                           ]]\n","display(gdsc_subset.head())"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"6jIqZyHZ2V-E","outputId":"4ad38435-2125-4655-95dc-3412b1450744"},"outputs":[{"data":{"text/html":"<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>COSMIC_ID</th>\n      <th>GDSC Tissue descriptor 1</th>\n      <th>GDSC Tissue descriptor 2</th>\n      <th>Cancer Type (matching TCGA label)</th>\n      <th>Microsatellite instability Status (MSI)</th>\n      <th>Screen Medium</th>\n      <th>CNA</th>\n      <th>Methylation</th>\n      <th>TARGET_PATHWAY</th>\n      <th>DRUG_NAME</th>\n      <th>LN_IC50</th>\n      <th>PUTATIVE_TARGET</th>\n      <th>PATHWAY_NAME</th>\n      <th>MIN_CONC</th>\n      <th>MAX_CONC</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>683667</td>\n      <td>nervous_system</td>\n      <td>medulloblastoma</td>\n      <td>MB</td>\n      <td>MSS/MSI-L</td>\n      <td>R</td>\n      <td>Y</td>\n      <td>Y</td>\n      <td>DNA replication</td>\n      <td>Camptothecin</td>\n      <td>-1.463887</td>\n      <td>TOP1</td>\n      <td>DNA replication</td>\n      <td>0.000100</td>\n      <td>0.1000</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>683667</td>\n      <td>nervous_system</td>\n      <td>medulloblastoma</td>\n      <td>MB</td>\n      <td>MSS/MSI-L</td>\n      <td>R</td>\n      <td>Y</td>\n      <td>Y</td>\n      <td>DNA replication</td>\n      <td>Camptothecin</td>\n      <td>-1.463887</td>\n      <td>Microtubule destabiliser</td>\n      <td>Mitosis</td>\n      <td>0.000100</td>\n      <td>0.1000</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>683667</td>\n      <td>nervous_system</td>\n      <td>medulloblastoma</td>\n      <td>MB</td>\n      <td>MSS/MSI-L</td>\n      <td>R</td>\n      <td>Y</td>\n      <td>Y</td>\n      <td>DNA replication</td>\n      <td>Camptothecin</td>\n      <td>-1.463887</td>\n      <td>DNA crosslinker</td>\n      <td>DNA replication</td>\n      <td>0.004002</td>\n      <td>6.0000</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>683667</td>\n      <td>nervous_system</td>\n      <td>medulloblastoma</td>\n      <td>MB</td>\n      <td>MSS/MSI-L</td>\n      <td>R</td>\n      <td>Y</td>\n      <td>Y</td>\n      <td>DNA replication</td>\n      <td>Camptothecin</td>\n      <td>-1.463887</td>\n      <td>Antimetabolite</td>\n      <td>Other</td>\n      <td>0.002001</td>\n      <td>2.0000</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>683667</td>\n      <td>nervous_system</td>\n      <td>medulloblastoma</td>\n      <td>MB</td>\n      <td>MSS/MSI-L</td>\n      <td>R</td>\n      <td>Y</td>\n      <td>Y</td>\n      <td>DNA replication</td>\n      <td>Camptothecin</td>\n      <td>-1.463887</td>\n      <td>Microtubule stabiliser</td>\n      <td>Mitosis</td>\n      <td>0.000013</td>\n      <td>0.0125</td>\n    </tr>\n  </tbody>\n</table>\n</div>","text/plain":"   COSMIC_ID GDSC Tissue descriptor 1 GDSC Tissue descriptor 2  \\\n0     683667           nervous_system          medulloblastoma   \n1     683667           nervous_system          medulloblastoma   \n2     683667           nervous_system          medulloblastoma   \n3     683667           nervous_system          medulloblastoma   \n4     683667           nervous_system          medulloblastoma   \n\n  Cancer Type (matching TCGA label) Microsatellite instability Status (MSI)  \\\n0                                MB                               MSS/MSI-L   \n1                                MB                               MSS/MSI-L   \n2                                MB                               MSS/MSI-L   \n3                                MB                               MSS/MSI-L   \n4                                MB                               MSS/MSI-L   \n\n  Screen Medium CNA Methylation   TARGET_PATHWAY     DRUG_NAME   LN_IC50  \\\n0             R   Y           Y  DNA replication  Camptothecin -1.463887   \n1             R   Y           Y  DNA replication  Camptothecin -1.463887   \n2             R   Y           Y  DNA replication  Camptothecin -1.463887   \n3             R   Y           Y  DNA replication  Camptothecin -1.463887   \n4             R   Y           Y  DNA replication  Camptothecin -1.463887   \n\n            PUTATIVE_TARGET     PATHWAY_NAME  MIN_CONC  MAX_CONC  \n0                      TOP1  DNA replication  0.000100    0.1000  \n1  Microtubule destabiliser          Mitosis  0.000100    0.1000  \n2           DNA crosslinker  DNA replication  0.004002    6.0000  \n3            Antimetabolite            Other  0.002001    2.0000  \n4    Microtubule stabiliser          Mitosis  0.000013    0.0125  "},"metadata":{},"output_type":"display_data"}],"source":["data = pd.merge(gdsc_subset,gdsc2_subset, on='COSMIC_ID', how='left')\n","display(data.head())"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"X9bXedtx2V-E","outputId":"3ecc5516-ae85-4b05-dc6d-1e377d6b7794"},"outputs":[{"name":"stdout","output_type":"stream","text":["0\n","0\n","0\n","0\n","0\n","0\n","0\n","0\n","0\n","0\n","0\n","0\n","0\n","0\n","0\n"]}],"source":["data = data.dropna()\n","for col in data.columns:\n","    print(data[col].isnull().sum())"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"l_lEj6hZ2V-E","outputId":"641fecf4-6f3d-4c04-bf5d-b2d031d29fac"},"outputs":[{"data":{"text/html":"<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>COSMIC_ID</th>\n      <th>GDSC Tissue descriptor 1</th>\n      <th>GDSC Tissue descriptor 2</th>\n      <th>Cancer Type (matching TCGA label)</th>\n      <th>Microsatellite instability Status (MSI)</th>\n      <th>Screen Medium</th>\n      <th>CNA</th>\n      <th>Methylation</th>\n      <th>TARGET_PATHWAY</th>\n      <th>DRUG_NAME</th>\n      <th>LN_IC50</th>\n      <th>PUTATIVE_TARGET</th>\n      <th>PATHWAY_NAME</th>\n      <th>MIN_CONC</th>\n      <th>MAX_CONC</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>62505405</th>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n    </tr>\n    <tr>\n      <th>62505406</th>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n    </tr>\n    <tr>\n      <th>62505407</th>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n    </tr>\n    <tr>\n      <th>62505408</th>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n    </tr>\n    <tr>\n      <th>62505409</th>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n      <td>False</td>\n    </tr>\n  </tbody>\n</table>\n<p>42811384 rows × 15 columns</p>\n</div>","text/plain":"          COSMIC_ID  GDSC Tissue descriptor 1  GDSC Tissue descriptor 2  \\\n0             False                     False                     False   \n1             False                     False                     False   \n2             False                     False                     False   \n3             False                     False                     False   \n4             False                     False                     False   \n...             ...                       ...                       ...   \n62505405      False                     False                     False   \n62505406      False                     False                     False   \n62505407      False                     False                     False   \n62505408      False                     False                     False   \n62505409      False                     False                     False   \n\n          Cancer Type (matching TCGA label)  \\\n0                                     False   \n1                                     False   \n2                                     False   \n3                                     False   \n4                                     False   \n...                                     ...   \n62505405                              False   \n62505406                              False   \n62505407                              False   \n62505408                              False   \n62505409                              False   \n\n          Microsatellite instability Status (MSI)  Screen Medium    CNA  \\\n0                                           False          False  False   \n1                                           False          False  False   \n2                                           False          False  False   \n3                                           False          False  False   \n4                                           False          False  False   \n...                                           ...            ...    ...   \n62505405                                    False          False  False   \n62505406                                    False          False  False   \n62505407                                    False          False  False   \n62505408                                    False          False  False   \n62505409                                    False          False  False   \n\n          Methylation  TARGET_PATHWAY  DRUG_NAME  LN_IC50  PUTATIVE_TARGET  \\\n0               False           False      False    False            False   \n1               False           False      False    False            False   \n2               False           False      False    False            False   \n3               False           False      False    False            False   \n4               False           False      False    False            False   \n...               ...             ...        ...      ...              ...   \n62505405        False           False      False    False            False   \n62505406        False           False      False    False            False   \n62505407        False           False      False    False            False   \n62505408        False           False      False    False            False   \n62505409        False           False      False    False            False   \n\n          PATHWAY_NAME  MIN_CONC  MAX_CONC  \n0                False     False     False  \n1                False     False     False  \n2                False     False     False  \n3                False     False     False  \n4                False     False     False  \n...                ...       ...       ...  \n62505405         False     False     False  \n62505406         False     False     False  \n62505407         False     False     False  \n62505408         False     False     False  \n62505409         False     False     False  \n\n[42811384 rows x 15 columns]"},"metadata":{},"output_type":"display_data"}],"source":["display(data.isnull())"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"4g4f2pZW2V-F","outputId":"b1e371b7-a130-4f1a-83eb-104c8fdf7c9c"},"outputs":[{"data":{"text/plain":"array(['nervous_system', 'skin', 'urogenital_system', 'lung_NSCLC',\n       'lung_SCLC', 'lung', 'neuroblastoma', 'myeloma', 'pancreas',\n       'aero_dig_tract', 'breast', 'kidney', 'leukemia',\n       'large_intestine', 'thyroid', 'digestive_system', 'lymphoma'],\n      dtype=object)"},"metadata":{},"output_type":"display_data"}],"source":["display(data['GDSC Tissue descriptor 1'].unique())"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"excq_x2G2V-F","outputId":"9921126c-7b3a-4923-fe71-d0196c34b123"},"outputs":[{"name":"stdout","output_type":"stream","text":["Requirement already satisfied: scikit-learn in /opt/conda/lib/python3.10/site-packages (1.2.2)\n","Requirement already satisfied: numpy>=1.17.3 in /opt/conda/lib/python3.10/site-packages (from scikit-learn) (1.26.4)\n","Requirement already satisfied: scipy>=1.3.2 in /opt/conda/lib/python3.10/site-packages (from scikit-learn) (1.14.1)\n","Requirement already satisfied: joblib>=1.1.1 in /opt/conda/lib/python3.10/site-packages (from scikit-learn) (1.4.2)\n","Requirement already satisfied: threadpoolctl>=2.0.0 in /opt/conda/lib/python3.10/site-packages (from scikit-learn) (3.5.0)\n"]}],"source":["!pip install scikit-learn"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"5mF7Pq_c2V-F","outputId":"3f1b7b8f-e8df-448c-c71f-b00cf5a0d97f"},"outputs":[{"data":{"text/html":"<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>COSMIC_ID</th>\n      <th>GDSC Tissue descriptor 1</th>\n      <th>GDSC Tissue descriptor 2</th>\n      <th>Cancer Type (matching TCGA label)</th>\n      <th>Microsatellite instability Status (MSI)</th>\n      <th>Screen Medium</th>\n      <th>CNA</th>\n      <th>Methylation</th>\n      <th>TARGET_PATHWAY</th>\n      <th>DRUG_NAME</th>\n      <th>LN_IC50</th>\n      <th>PUTATIVE_TARGET</th>\n      <th>PATHWAY_NAME</th>\n      <th>MIN_CONC</th>\n      <th>MAX_CONC</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>683667</td>\n      <td>11</td>\n      <td>24</td>\n      <td>18</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>7</td>\n      <td>73</td>\n      <td>-1.463887</td>\n      <td>165</td>\n      <td>7</td>\n      <td>0.000100</td>\n      <td>0.1000</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>683667</td>\n      <td>11</td>\n      <td>24</td>\n      <td>18</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>7</td>\n      <td>73</td>\n      <td>-1.463887</td>\n      <td>113</td>\n      <td>15</td>\n      <td>0.000100</td>\n      <td>0.1000</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>683667</td>\n      <td>11</td>\n      <td>24</td>\n      <td>18</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>7</td>\n      <td>73</td>\n      <td>-1.463887</td>\n      <td>49</td>\n      <td>7</td>\n      <td>0.004002</td>\n      <td>6.0000</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>683667</td>\n      <td>11</td>\n      <td>24</td>\n      <td>18</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>7</td>\n      <td>73</td>\n      <td>-1.463887</td>\n      <td>18</td>\n      <td>16</td>\n      <td>0.002001</td>\n      <td>2.0000</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>683667</td>\n      <td>11</td>\n      <td>24</td>\n      <td>18</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>1</td>\n      <td>7</td>\n      <td>73</td>\n      <td>-1.463887</td>\n      <td>114</td>\n      <td>15</td>\n      <td>0.000013</td>\n      <td>0.0125</td>\n    </tr>\n  </tbody>\n</table>\n</div>","text/plain":"   COSMIC_ID  GDSC Tissue descriptor 1  GDSC Tissue descriptor 2  \\\n0     683667                        11                        24   \n1     683667                        11                        24   \n2     683667                        11                        24   \n3     683667                        11                        24   \n4     683667                        11                        24   \n\n   Cancer Type (matching TCGA label)  Microsatellite instability Status (MSI)  \\\n0                                 18                                        1   \n1                                 18                                        1   \n2                                 18                                        1   \n3                                 18                                        1   \n4                                 18                                        1   \n\n   Screen Medium  CNA  Methylation  TARGET_PATHWAY  DRUG_NAME   LN_IC50  \\\n0              1    1            1               7         73 -1.463887   \n1              1    1            1               7         73 -1.463887   \n2              1    1            1               7         73 -1.463887   \n3              1    1            1               7         73 -1.463887   \n4              1    1            1               7         73 -1.463887   \n\n   PUTATIVE_TARGET  PATHWAY_NAME  MIN_CONC  MAX_CONC  \n0              165             7  0.000100    0.1000  \n1              113            15  0.000100    0.1000  \n2               49             7  0.004002    6.0000  \n3               18            16  0.002001    2.0000  \n4              114            15  0.000013    0.0125  "},"metadata":{},"output_type":"display_data"}],"source":["from sklearn.preprocessing import LabelEncoder\n","\n","label_encoder = LabelEncoder()\n","\n","for col in data.columns:\n","    if data[col].dtype == 'object':\n","        data[col] = label_encoder.fit_transform(data[col])\n","\n","display(data.head())"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"KKM5I39u2V-F"},"outputs":[],"source":["output_file = '/kaggle/working/processed_data.csv'\n","data.to_csv(output_file, index=False)"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"SmMAk7jX2V-F"},"outputs":[],"source":["x = data.drop(columns=['COSMIC_ID','LN_IC50'])\n","y = data['LN_IC50']\n","output_x = '/kaggle/working/x.csv'\n","output_y = '/kaggle/working/y.csv'\n","x.to_csv(output_x, index=False)\n","y.to_csv(output_y, index=False)"]},{"cell_type":"markdown","metadata":{"id":"iTddPdUL2-Ji"},"source":["# Model - XGBoost"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"3wqqiRYR3CDv"},"outputs":[],"source":["import pandas as pd\n","import numpy as np\n","import os\n","import joblib\n","import matplotlib.pyplot as plt\n","import gc\n","\n","import xgboost as xgb\n","from sklearn.model_selection import train_test_split\n","from sklearn.preprocessing import StandardScaler\n","from sklearn.metrics import mean_squared_error, r2_score"]},{"cell_type":"code","execution_count":null,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":2372,"status":"ok","timestamp":1733390279891,"user":{"displayName":"Nana Yang","userId":"01362796942571968522"},"user_tz":-480},"id":"BMf14CWAAwsn","outputId":"70b77377-4810-41fd-ecee-8abce5475076"},"outputs":[{"output_type":"stream","name":"stdout","text":["Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount(\"/content/drive\", force_remount=True).\n"]}],"source":["from google.colab import drive\n","drive.mount('/content/drive')\n","\n","# 記得改成自己的路徑\n","base_path = '/content/drive/My Drive/NCU/04_deep_learning/final_project/data'\n","\n","# 創建保存模型的目錄\n","save_model_path = os.path.join(base_path, 'save_model')\n","os.makedirs(save_model_path, exist_ok=True)"]},{"cell_type":"markdown","metadata":{"id":"9EMoLd503RBH"},"source":["### Load Data"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"QqznGJnZ3T7-"},"outputs":[],"source":["def load_data(x_path: str, y_path: str):\n","    \"\"\"\n","    Load the preprocessed data from CSV files.\n","\n","    Args:\n","        x_path: Path to features CSV file\n","        y_path: Path to target CSV file\n","\n","    Returns:\n","        Tuple containing features DataFrame and target array\n","    \"\"\"\n","    X = pd.read_csv(x_path)\n","    y = pd.read_csv(y_path)\n","    return X, y.values.ravel()"]},{"cell_type":"markdown","metadata":{"id":"RgxXsgrl6ptp"},"source":["### 分割資料"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"C8a_fdPJ6rPV"},"outputs":[],"source":["def split_data(X: pd.DataFrame, y: np.ndarray, test_size: float = 0.2, random_state: int = 42):\n","    \"\"\"\n","    Split data into training and testing sets.\n","\n","    Args:\n","        X: Feature DataFrame\n","        y: Target array\n","        test_size: Proportion of data to use for testing\n","        random_state: Random seed for reproducibility\n","\n","    Returns:\n","        Tuple containing train and test splits\n","    \"\"\"\n","    return train_test_split(X, y, test_size=test_size, random_state=random_state)"]},{"cell_type":"markdown","metadata":{"id":"c0Nud1Fh3fwH"},"source":["### 特徵標準化"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"KtbEqdb43fTt"},"outputs":[],"source":["def process_features(X, n_components=0.95):\n","    \"\"\"\n","    特徵處理：標準化和降維\n","\n","    Parameters:\n","        X: 特徵矩陣\n","        n_components: PCA 保留的變異比例\n","\n","    Returns:\n","        tuple: (處理後的特徵矩陣, scaler, pca)\n","    \"\"\"\n","    # 標準化\n","    scaler = StandardScaler()\n","    X_scaled = scaler.fit_transform(X)\n","\n","    gc.collect()  # 清理記憶體\n","    return X_scaled, scaler"]},{"cell_type":"markdown","metadata":{"id":"xbOlQ9K_4JfE"},"source":["### SVM 模型建立"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"YmRMFArA4LDI"},"outputs":[],"source":["def create_xgb_model(n_estimators=150, learning_rate=0.01, max_depth=8, random_state=42):\n","    xgb_model = xgb.XGBRegressor (\n","      objective = 'reg:squarederror',  # 均方誤差作為損失函數\n","      n_estimators = n_estimators,  # 樹的數量\n","      learning_rate = learning_rate,  # 學習率\n","      max_depth = max_depth,  # 樹的最大深度\n","      random_state = random_state\n","    )\n","\n","    return xgb_model"]},{"cell_type":"markdown","metadata":{"id":"pWhGXx5N6Dio"},"source":["### 預測及評估模型效能"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"DhKZtwZL6DQ_"},"outputs":[],"source":["def model_predict(model, X_train, X_test, y_train, y_test):\n","    \"\"\"\n","    Evaluate model performance on train and test sets.\n","\n","    Args:\n","        model: Trained model\n","        X_train: Training features\n","        X_test: Testing features\n","        y_train: Training targets\n","        y_test: Testing targets\n","\n","    Returns:\n","        Dictionary containing performance metrics\n","    \"\"\"\n","    y_train_pred = model.predict(X_train)\n","    y_test_pred = model.predict(X_test)\n","\n","    return {\n","        'train_mse': mean_squared_error(y_train, y_train_pred),\n","        'test_mse': mean_squared_error(y_test, y_test_pred),\n","        'train_r2': r2_score(y_train, y_train_pred),\n","        'test_r2': r2_score(y_test, y_test_pred),\n","        'y_train_pred': y_train_pred,\n","        'y_test_pred': y_test_pred\n","    }"]},{"cell_type":"markdown","metadata":{"id":"CQSRpvNF-yUE"},"source":["### 繪圖"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"3dDfAw9E-yGC"},"outputs":[],"source":["def plot_predictions(y_train: np.ndarray, y_test: np.ndarray,\n","                    y_train_pred: np.ndarray, y_test_pred: np.ndarray):\n","    \"\"\"\n","    Visualize model predictions.\n","\n","    Args:\n","        y_train: Actual training targets\n","        y_test: Actual testing targets\n","        y_train_pred: Predicted training targets\n","        y_test_pred: Predicted testing targets\n","    \"\"\"\n","    plt.figure(figsize=(12, 5))\n","\n","    # Training set predictions\n","    plt.subplot(1, 2, 1)\n","    plt.scatter(y_train, y_train_pred, alpha=0.5)\n","    plt.plot([y_train.min(), y_train.max()], [y_train.min(), y_train.max()], 'r--', lw=2)\n","    plt.xlabel('Actual LN_IC50')\n","    plt.ylabel('Predicted LN_IC50')\n","    plt.title('Training Set Predictions')\n","\n","    # Test set predictions\n","    plt.subplot(1, 2, 2)\n","    plt.scatter(y_test, y_test_pred, alpha=0.5)\n","    plt.plot([y_test.min(), y_test.max()], [y_test.min(), y_test.max()], 'r--', lw=2)\n","    plt.xlabel('Actual LN_IC50')\n","    plt.ylabel('Predicted LN_IC50')\n","    plt.title('Test Set Predictions')\n","\n","    plt.tight_layout()\n","    plt.show()"]},{"cell_type":"markdown","metadata":{"id":"Wa5zjksD-3Pr"},"source":["### Save model"]},{"cell_type":"code","execution_count":null,"metadata":{"id":"iXaMCQRv-3E4"},"outputs":[],"source":["def save_model(model, scaler, metrics, model_path, scaler_path):\n","    \"\"\"\n","    Save model, scaler and metrics to files.\n","\n","    Args:\n","        model: Traine model\n","        scaler: Fitted StandardScaler\n","        metrics: Dictionary of performance metrics\n","        model_path: Path to save model\n","        scaler_path: Path to save scaler\n","    \"\"\"\n","    joblib.dump(model, model_path)\n","    joblib.dump(scaler, scaler_path)\n","    print(\"Model and scaler saved successfully\")\n","    print(\"\\nModel Performance Metrics:\")\n","    for metric, value in metrics.items():\n","        if isinstance(value, float):\n","            print(f\"{metric}: {value:.4f}\")"]},{"cell_type":"markdown","metadata":{"id":"8t4gKAO7-7aF"},"source":["## Main fucntion"]},{"cell_type":"code","execution_count":null,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"_nmdI94s-7TU","outputId":"7e890a3c-7c32-4962-b752-fcbdbc3e6668"},"outputs":[{"output_type":"stream","name":"stdout","text":["data loading...\n","data spliting...\n","X_train shape: (34249107, 13)\n","X_test shape: (8562277, 13)\n","data scaling...\n"]}],"source":["def main() -> None:\n","    # Load data\n","    print(\"data loading...\")\n","    X, y = load_data(\n","        os.path.join(base_path, 'x.csv'),\n","        os.path.join(base_path, 'y.csv')\n","    )\n","\n","    # Split data\n","    print(\"data spliting...\")\n","    X_train, X_test, y_train, y_test = split_data(X, y)\n","    print(f\"---X_train shape: {X_train.shape}\")\n","    print(f\"---X_test shape: {X_test.shape}\")\n","\n","    # 特徵處理\n","    print(\"data scaling...\")\n","    X_train_processed, scaler, pca = process_features(X_train)\n","    X_test_processed = pca.transform(scaler.transform(X_test))\n","    print(f\"---X_train_processed shape: {X_train_processed.shape}\")\n","    print(f\"---X_test_processed shape: {X_test_processed.shape}\")\n","\n","    # Create and train model\n","    print(\"***Stast model training...\")\n","\n","    # 要另外調整參數的話可以用這個:\n","    '''\n","      model_params = {\n","        objective = 'reg:squarederror',\n","        n_estimators = 100,\n","        learning_rate = 0.1,\n","        max_depth = 6,\n","        random_state = 42\n","    }\n","\n","    model = create_xgb_model(**model_params)\n","    '''\n","\n","    model = create_xgb_model()\n","    xgb_model.fit(X_train, y_train)\n","    print(\"model trained successfully\")\n","\n","    # 預測與評估\n","    print(\"model evaluating...\")\n","    metrics = model_predict(trained_model, X_train_processed, X_test_processed, y_train, y_test)\n","    print(\"model evaluated successfully\")\n","    print(\"\\n---Model Performance Metrics:\")\n","    for metric, value in metrics.items():\n","        if isinstance(value, float):\n","            print(f\"{metric}: {value:.4f}\")\n","\n","    # Plot results\n","    plot_predictions(y_train, y_test, metrics['y_train_pred'], metrics['y_test_pred'])\n","\n","    # Save model and results\n","    save_model(\n","        trained_model,\n","        scaler,\n","        metrics,\n","        os.path.join(save_model_path, 'xgb_model.joblib'),\n","        os.path.join(save_model_path, 'scaler.joblib')\n","    )\n","    print(\"model saved successfully\")\n","\n","if __name__ == \"__main__\":\n","    main()"]},{"cell_type":"markdown","metadata":{"id":"xUPraf_R--NK"},"source":["## (預留) 如果要使用到 saved model"]},{"cell_type":"code","source":["# def load_saved_model(model_path: str, scaler_path: str):\n","#     \"\"\"\n","#     載入已儲存的模型和scaler\n","\n","#     Args:\n","#         model_path: 模型檔案路徑\n","#         scaler_path: scaler檔案路徑\n","#     \"\"\"\n","#     # 載入模型和scaler\n","#     model = joblib.load(model_path)\n","#     scaler = joblib.load(scaler_path)\n","#     return model, scaler"],"metadata":{"id":"tr-lWvaPGsyA"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["# def make_predictions(model, scaler, new_data: pd.DataFrame):\n","#     \"\"\"\n","#     使用載入的模型進行預測\n","\n","#     Args:\n","#         model: 載入的SVR模型\n","#         scaler: 載入的StandardScaler\n","#         new_data: 要預測的新數據\n","#     \"\"\"\n","#     # 使用相同的scaler轉換新數據\n","#     scaled_data = scaler.transform(new_data)\n","\n","#     # 使用模型進行預測\n","#     predictions = model.predict(scaled_data)\n","#     return predictions"],"metadata":{"id":"3Ih247pmGuEk"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["# def main():\n","#     # 1. 載入模型和scaler\n","#     model, scaler = load_saved_model(\n","#         f'{save_model_path}/svm_model.joblib',\n","#         f'{save_model_path}/scaler.joblib'\n","#     )\n","\n","#     # 2. 準備新數據 (這裡要確保新數據的格式和訓練時相同)\n","#     new_data = pd.read_csv(f'{base_path}/new_data.csv')  # 替換成你的新數據路徑\n","\n","#     # 3. 進行預測\n","#     predictions = make_predictions(model, scaler, new_data)\n","\n","#     # 4. 查看預測結果\n","#     print(\"Predictions:\", predictions)\n","\n","#     # 5. 如果需要，可以將預測結果保存到CSV\n","#     results_df = pd.DataFrame({\n","#         'Predictions': predictions\n","#     })\n","#     results_df.to_csv('predictions.csv', index=False)\n","\n","# if __name__ == \"__main__\":\n","#     main()"],"metadata":{"id":"IIf_wABWGxLT"},"execution_count":null,"outputs":[]}],"metadata":{"colab":{"provenance":[],"toc_visible":true,"collapsed_sections":["xUPraf_R--NK"]},"kaggle":{"accelerator":"none","dataSources":[{"datasetId":5535401,"sourceId":9167838,"sourceType":"datasetVersion"}],"dockerImageVersionId":30804,"isGpuEnabled":false,"isInternetEnabled":true,"language":"python","sourceType":"notebook"},"kernelspec":{"display_name":"Python 3","name":"python3"},"language_info":{"codemirror_mode":{"name":"ipython","version":3},"file_extension":".py","mimetype":"text/x-python","name":"python","nbconvert_exporter":"python","pygments_lexer":"ipython3","version":"3.10.14"}},"nbformat":4,"nbformat_minor":0}